spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/SecretsPropertySourceLocator.java

```diff
@@ -61,22 +70,23 @@ public SecretsPropertySourceLocator(KubernetesClient client,
 	}
 
 	@Override
-	public PropertySource locate(Environment environment) {
+	public PropertySource<?> locate(Environment environment) {
 		if (environment instanceof ConfigurableEnvironment) {
 			ConfigurableEnvironment env = (ConfigurableEnvironment) environment;
 
 			List<SecretsConfigProperties.NormalizedSource> sources = this.properties
 					.determineSources();
 			CompositePropertySource composite = new CompositePropertySource(
 					"composite-secrets");
-			if (this.properties.isEnableApi()) {
```

Comment:
the previous code would make the return from `getKubernetesPropertySourceForSingleSecret` to be the _last_ one in `composite`; and that is the only guarantee that existed here. 

To maintain it, I `putPathConfig(composite);` before it; and also changed to `composite.addPropertySource` instead of `composite.addFirstPropertySource`